bundles/org.openhab.binding.airgradient/src/main/java/org/openhab/binding/airgradient/internal/handler/AirGradientAPIHandler.java

```diff
@@ -0,0 +1,251 @@
+        }
+
+        return results;
+    }
+
+    /**
+     * Return list of measures from AirGradient API.
+     *
+     * @return list of measures
+     */
+    public List<Measure> getMeasures() {
+        ContentResponse response = sendRequest(
+                RESTHelper.generateRequest(httpClient, RESTHelper.generateMeasuresUrl(apiConfig)));
+        if (response != null) {
+            String contentType = response.getMediaType();
+            logger.debug("Got measurements with status {}: {} ({})", response.getStatus(),
+                    response.getContentAsString(), contentType);
+
+            if (RESTHelper.isSuccess(response)) {
+                String stringResponse = response.getContentAsString().trim();
+
+                if (null != contentType)
+                    switch (contentType) {
+                        case CONTENTTYPE_JSON:
+                            return JsonParserHelper.parseJson(gson, stringResponse);
+                        case CONTENTTYPE_TEXT:
+                            return PrometheusParserHelper.parsePrometheus(stringResponse);
+                        case CONTENTTYPE_OPENMETRICS:
+                            return PrometheusParserHelper.parsePrometheus(stringResponse);
+                        default:
+                            logger.debug("Unhandled content type returned: {}", contentType);
+                    }
+            }
+        }
+
+        return Collections.emptyList();
+    }
+
+    public void setLedMode(String serialNo, String mode) {
+        Request request = httpClient.newRequest(RESTHelper.generateGetLedsModeUrl(apiConfig, serialNo));
+        request.timeout(REQUEST_TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+        request.method(HttpMethod.PUT);
+        request.header(HttpHeader.CONTENT_TYPE, CONTENTTYPE_JSON);
+        LedMode ledMode = new LedMode();
+        ledMode.mode = mode;
+        String modeJson = gson.toJson(ledMode);
+        logger.debug("Setting LEDS mode for {}: {}", serialNo, modeJson);
+        request.content(new StringContentProvider(CONTENTTYPE_JSON, modeJson, StandardCharsets.UTF_8));
+        sendRequest(request);
+    }
+
+    public void calibrateCo2(String serialNo) {
+        logger.debug("Triggering CO2 calibration for {}", serialNo);
+        sendRequest(RESTHelper.generateRequest(httpClient, RESTHelper.generateCalibrationCo2Url(apiConfig, serialNo),
+                HttpMethod.POST));
+    }
+
+    private @Nullable ContentResponse sendRequest(@Nullable final Request request) {
+        if (request == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Unable to generate request");
+        }
+
+        @Nullable
+        ContentResponse response = null;
+        try {
+            response = request.send();
+            if (response != null) {
+                logger.debug("Response from {}: {}", request.getURI(), response.getStatus());
+                if (RESTHelper.isSuccess(response)) {
```

Comment:
Any reason to create a cusom method while `response.isSuccess()` is available?

Reply:
I was looking for it, but couldn't find it neither in Jetty 9 https://eclipse.dev/jetty/javadoc/jetty-9/org/eclipse/jetty/client/api/ContentResponse.html nor in Jetty 12 https://eclipse.dev/jetty/javadoc/jetty-12/org/eclipse/jetty/client/ContentResponse.html  Am I using the wrong http client? (I use factory.getCommonHttpClient() in the factory to get it).

Reply:
You are close. One level higher. https://eclipse.dev/jetty/javadoc/jetty-9/org/eclipse/jetty/client/api/Result.html has the methods to check if it succeeded

Reply:
Thanks! Found the jetty library for it. My C# background made me look for the method on the response object.